pydantic_ai_slim/pydantic_ai/messages.py

```diff
@@ -99,6 +113,9 @@ class FileUrl(ABC):
     * If False, the URL is sent directly to the model and no download is performed.
     """
 
+    vendor_metadata: VendorMetadata | None = None
```

Comment:
What do you think about making this just a `dict[str, Any]` for now that is passed to the API directly without needing to map the keys? 

If we go the route of using an object, that'd be a lot like `ModelSettings` so it would need to be a `TypedDict` with vendor-specific subclasses with prefixed fields. Then we'd likely also want to make it video-specific so that the suggested autocomplete options make sense in context. So I think for now just a dict would be more flexible :)

Reply:
Changed it to dict[str, Any] and added an exact clarification in the description.